benchmarks/offload.py

```diff
@@ -57,37 +79,42 @@ def train_epoch(args):
         for batch_inputs, batch_outputs in dataloader:
             batch_inputs, batch_outputs = batch_inputs.to("cuda"), batch_outputs.to("cuda")
             start = time.time_ns()
-            with torch.autograd.profiler.profile(use_cuda=True, profile_memory=True) as prof:
+            with _get_profiler_context() as prof:
                 optimizer.zero_grad()
                 inputs = batch_inputs.reshape(-1, args.inputs * args.inputs)
-                with torch.autograd.profiler.record_function("model_training"):
-                    output = model(inputs)
-                    loss = criterion(output, target=batch_outputs)
-                    loss.backward()
+                with _get_profiler_record_context("model_training"):
+                    with _get_fp16_context(use_fp16=args.use_fp16):
+                        output = model(inputs)
+                        loss = criterion(output, target=batch_outputs)
+                        loss.backward()
                     optimizer.step()
-            prof.export_chrome_trace("/tmp/mpi_prof")
-            logging.info(f"Memory table {prof.key_averages().table()}")
-            logging.info("Memory stats are " + str(torch.cuda.memory_stats(0)["allocated_bytes.all.peak"] / 2 ** 30))
+            logging.info(
+                "Memory stats are {:.2f}GB".format(torch.cuda.memory_stats(0)["allocated_bytes.all.peak"] / 2 ** 30)
```

Comment:
dummy thought earlier in the day: it would be great to add some ballpark computation of the expected size at some point (given the batch size + model/shards), just for comparison